src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/TreeViewTests.cs

```diff
@@ -0,0 +1,761 @@
+				treeViewNode1.Children.Add(treeViewNode5);
+
+				treeView = new TreeView();
+				TestServices.WindowHelper.WindowContent = treeView;
+			});
+
+			await TestServices.WindowHelper.WaitForIdle();
+
+			await RunOnUIThread.Execute(() =>
+			{
+
+				var listControl = FindVisualChildByName(treeView, "ListControl") as TreeViewList;
+				treeView.RootNodes.Add(treeViewNode1);
+				Verify.AreEqual(listControl.GetItems()?.Count(), 1);
+
+				treeView.Expand(treeViewNode1);
+				Verify.AreEqual(listControl.GetItems()?.Count(), 5);
+
+				treeViewNode1.Children.RemoveAt(2);
+				Verify.AreEqual(listControl.GetItems()?.Count(), 4);
+				Verify.AreEqual(listControl.GetItemFromIndex(0), treeViewNode1);
+				Verify.AreEqual(listControl.GetItemFromIndex(1), treeViewNode2);
+				Verify.AreEqual(listControl.GetItemFromIndex(2), treeViewNode3);
+				Verify.AreEqual(listControl.GetItemFromIndex(3), treeViewNode5);
+
+				treeViewNode1.Children.Insert(1, treeViewNode4);
+				Verify.AreEqual(listControl.GetItems()?.Count(), 5);
+				Verify.AreEqual(listControl.GetItemFromIndex(0), treeViewNode1);
+				Verify.AreEqual(listControl.GetItemFromIndex(1), treeViewNode2);
+				Verify.AreEqual(listControl.GetItemFromIndex(2), treeViewNode4);
+				Verify.AreEqual(listControl.GetItemFromIndex(3), treeViewNode3);
+				Verify.AreEqual(listControl.GetItemFromIndex(4), treeViewNode5);
+
+				treeViewNode1.Children.Clear();
+				Verify.AreEqual(listControl.GetItems()?.Count(), 1);
+				Verify.AreEqual(listControl.GetItemFromIndex(0), treeViewNode1);
+			});
+
+			await TestServices.WindowHelper.WaitForIdle();
+		}
+
+		//[TestMethod] Disabled with issue number #1775
+		public void TreeViewInheritanceTest()
+		{
+			StackPanel stackPanel = new StackPanel();
+			SpecialTreeView inheritedTreeView = new SpecialTreeView();
+			SpecialTreeViewList inheritedTreeViewList = new SpecialTreeViewList();
+			SpecialTreeViewItem inheritedTreeViewItem = new SpecialTreeViewItem();
+			SpecialTreeViewNode inheritedTreeViewNode = new SpecialTreeViewNode();
+			IList<string> data = Enumerable.Range(0, 5).Select(x => "Item " + x).ToList();
+
+			Verify.IsNotNull(stackPanel);
+			stackPanel.Children.Add(inheritedTreeView);
+			stackPanel.Children.Add(inheritedTreeViewList);
+			inheritedTreeViewList.ItemsSource = data;
+			TestServices.WindowHelper.WindowContent = stackPanel;
+			stackPanel.UpdateLayout();
+		}
+
+		[TestMethod]
+		public async Task VerifyTreeViewIsNotTabStop()
+		{
+			TreeView treeView = null;
+			await RunOnUIThread.Execute(() =>
+			{
+				treeView = new TreeView();
+				TestServices.WindowHelper.WindowContent = treeView;
+			});
+
+			await TestServices.WindowHelper.WaitForIdle();
+
+			await RunOnUIThread.Execute(() =>
+			{
+				Verify.IsFalse(treeView.IsTabStop);
+			});
+
+			await TestServices.WindowHelper.WaitForIdle();
+		}
+
+		//[TestMethod]
+		//[RunsOnUIThread]
+		//public void VerifyClearingNodeWithNoChildren()
+		//{
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)